xbmc/video/test/TestVideoInfoScanner.cpp

```diff
@@ -52,37 +58,46 @@ static const TestEntry TestData[] = {
     {"foo 2x01x02.mkv", {{2, 1}, {2, 2}}, {{2, 1}, {2, 2}}},
     {"foo ep01-02.mkv", {{1, 1}, {1, 2}}, {{1, 1}, {1, 2}}},
     {"foo.S02E01S02E03.mkv", {{2, 1}, {2, 3}}, {{2, 1}, {2, 3}}},
-    // spanning episodes
-    {"foo.E03-05.mkv", {{1, 3}, {1, 4}, {1, 5}}, {{1, 3}, {1, 5}}},
-    {"foo.ep03-05.mkv", {{1, 3}, {1, 4}, {1, 5}}, {{1, 3}, {1, 5}}},
+    // spanning episode ranges
     {"foo.ep03-ep05.mkv", {{1, 3}, {1, 4}, {1, 5}}, {{1, 3}, {1, 5}}},
-    {"foo.5x03-5x05.mkv", {{5, 3}, {5, 4}, {5, 5}}, {{5, 3}, {5, 5}}},
-    {"foo.S05E03-05.mkv", {{5, 3}, {5, 4}, {5, 5}}, {{5, 3}, {5, 5}}},
     {"foo.S05E03-E05.mkv", {{5, 3}, {5, 4}, {5, 5}}, {{5, 3}, {5, 5}}},
     {"foo.S05E03-S05E05.mkv", {{5, 3}, {5, 4}, {5, 5}}, {{5, 3}, {5, 5}}},
     {"foo.S05E01-E03-E05.mkv", {{5, 1}, {5, 2}, {5, 3}, {5, 4}, {5, 5}}, {{5, 1}, {5, 3}, {5, 5}}},
     {"foo.S05E01-S05E03-S05E05.mkv",
      {{5, 1}, {5, 2}, {5, 3}, {5, 4}, {5, 5}},
      {{5, 1}, {5, 3}, {5, 5}}},
+    {"foo.5x03-5x05.mkv", {{5, 3}, {5, 4}, {5, 5}}, {{5, 3}, {5, 5}}},
+    // not ranges
+    {"foo.E03-05.mkv", {{1, 3}, {1, 5}}, {{1, 3}, {1, 5}}},
+    {"foo.ep03-05.mkv", {{1, 3}, {1, 5}}, {{1, 3}, {1, 5}}},
+    {"foo.S05E03-05.mkv", {{5, 3}, {5, 5}}, {{5, 3}, {5, 5}}},
     // multi-season
     {"foo.S00E100S05E03E04E05.mkv",
      {{0, 100}, {5, 3}, {5, 4}, {5, 5}},
      {{0, 100}, {5, 3}, {5, 4}, {5, 5}}},
     {"foo.S01E01E02S02E01.mkv", {{1, 1}, {1, 2}, {2, 1}}, {{1, 1}, {1, 2}, {2, 1}}},
-    {"foo.S00E01-04S05E03-E06.mkv",
+    {"foo.S00E01-E04S05E03-E06.mkv",
      {{0, 1}, {0, 2}, {0, 3}, {0, 4}, {5, 3}, {5, 4}, {5, 5}, {5, 6}},
      {{0, 1}, {0, 4}, {5, 3}, {5, 6}}},
-    // expected (partial) failure
+    {"foo s01e01-s01e04.season1.s02e01-s02e04.season2.mkv",
```

Comment:
you're creating work for yourself with that one lol. not a priority if something has to go to code a reasonable implementation.

Reply:
OK but why create a new pattern? There are too many as it is in my opinion. Unless there is a need for it?